### PR TITLE
Improve password confirmation for passwordless authentication

### LINKED ISSUE #20186: When asking for password confirmation, and password confirmation feature is disabled (eg. in SAML or LDAP), then explain they can proceed without entering a password
Currently, when using SAML (and soon with LDAP) users see this prompt:
image

The password field is disabled, and users can click Yes to continue without a password and it will work

But there is no indication this is the case, so it would be great to, in this case (where the password confirmation is disabled).

How could we solve this issue?

- is it possible to automatically click the YES button or even hide the popover all together when password confirmation is disabled?
- otherwise if not possible, we can show a message within the popover below the title eg.

> Attention: The password confirmation feature is temporarily disabled. You can still proceed with this action by clicking 'YES' without entering a password.

### PR DESCRIPTION
**Pull request by sgiehl**
### Description:

Some plugins provide a passwordless possibility to authenticate. Such as LoginLdap or LoginSAML. Various critical actions in Matomo currently require the password to be confirmed in the UI to continue.
For passwordless plugins this confirmation is shown as well, but it is possible to simply click `Yes`, without typing a password.
This process is not very user friendly.

This PR changes the password confirmation prompt. If a password is not needed at all, the password input won't be shown. Also the confirmation message is slightly adjusted, so it doesn't confuse people.

**Before:**

With default message:

image

With custom message:

image

**After:**

With default message:

image

With custom message:

image

fixes #20186 

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: plugins/CorePluginsAdmin/vue/src/PasswordConfirmation/PasswordConfirmation.vue [hunk 1]
```diff
@@ -8,9 +8,18 @@
   <div class="confirm-password-modal modal" ref="root">
     <div class="modal-content">
       <div class="modal-text">
-        <slot></slot>
+        <div ref="content"><slot></slot></div>
+        <h2 v-if="!slotHasContent && !requiresPasswordConfirmation">
```
Comment (michalkleiner) [inline plugins/CorePluginsAdmin/vue/src/PasswordConfirmation/PasswordConfirmation.vue]: to keep the order of the conditions in the v-if same for all h2 variations
```suggestion
        <h2 v-if="!requiresPasswordConfirmation && !slotHasContent">
```

File: plugins/UsersManager/vue/src/PagedUsersList/PagedUsersList.vue [hunk 2]
```diff
@@ -388,7 +388,6 @@
             `<strong>${affectedUsersCount}</strong>`,
           ))"
       ></h2>
-      <p>{{ translate('UsersManager_ConfirmWithPassword') }}</p>
```
Comment (michalkleiner) [inline plugins/UsersManager/vue/src/PagedUsersList/PagedUsersList.vue:391]: Can't comment on the lines below as they are not part of the change, but should the next `PasswordConfirmation` component with `v-model="showPasswordConfirmationForAnonymousAccess"` after this have `h2` headings instead of `h3`s? If we're touching things here we might as well adjust that?

**Comment by michalkleiner** [comment]
@sgiehl the code change looks good.
Can you suggest what is the easiest way how to test this locally? Which plugin is the best to test with and what setup it requires locally? Thanks!

File: plugins/UsersManager/vue/src/PagedUsersList/PagedUsersList.vue (hunk 2 above)
Comment (sgiehl) [reply plugins/UsersManager/vue/src/PagedUsersList/PagedUsersList.vue:391]: I guess the reason for using a h3 there, was that the text can be quite long and using h2, might break it into a lot of lines.

**Comment by sgiehl** [comment]
@michalkleiner the easiest way is to simply adjust this line in core code:
https://github.com/matomo-org/matomo/blob/d209158e6ee34dc4b7c50df98c078954a750979f/core/Piwik.php#L293
The plugins are using the event below to manipulate that value.

File: plugins/UsersManager/vue/src/PagedUsersList/PagedUsersList.vue (hunk 2 above)
Comment (michalkleiner) [reply plugins/UsersManager/vue/src/PagedUsersList/PagedUsersList.vue:391]: Right, ok. The correct way would be to still use h2 tag with a class modifier to make the text smaller/look like h3 but keep the semantics. All good, out of scope here.

**Comment by michalkleiner** [comment]
> @michalkleiner the easiest way is to simply adjust this line in core code:
> 
> https://github.com/matomo-org/matomo/blob/d209158e6ee34dc4b7c50df98c078954a750979f/core/Piwik.php#L293
> 
> 
> The plugins are using the event below to manipulate that value.

~Hmmm, even if I return false from that method, clear caches, log out and back in, I still get the password field.~

Was adjusting code for a wrong instance 🙈

**Review by michalkleiner — APPROVED** [review]
Works for me both password and password-less variants.

I'm not 100% convinced that _Confirm_/_Cancel_ is always better than _Yes_/_No_ with the current content/questions, but we can adjust the copy if needed so that the _Confirm_ action makes sense in all instances.

If the question is "Are you sure ... XYZ?" then _Yes_/_No_ seems better, but we can reword such instances to "Please confirm that you want XYZ"

Overall a good job, just one minor code suggestion for consistency.
